### PR TITLE
versionbump: golang 1.21.4

### PR DESCRIPTION
**Pull request by justinsb**
Bump to golang 1.21.4 in a few more places

### CODE REVIEW

File: .github/workflows/presubmit.yaml [hunk 1]
```diff
@@ -29,11 +28,11 @@ jobs:
       - uses: actions/checkout@v3
         with:
           # This is introduced to make sure 'go generate' works properly
-          path: 'src/k8s-config-connector'
+          path: "src/k8s-config-connector"
```
Comment (diviner524) [inline .github/workflows/presubmit.yaml:31]: Curious why we want to change to double quote?

**Comment by diviner524** [comment]
mock-tests seem to be failing?

    harness.go:152: error stopping envtest environment: timeout waiting for process kube-apiserver to stop

File: .github/workflows/presubmit.yaml (hunk 1 above)
Comment (justinsb) [reply .github/workflows/presubmit.yaml:31]: Ah, VSCode just does this automatically.  We'll be fighting the tools with single quotes.  I can split it out if desired.

**Comment by justinsb** [comment]
> timeout waiting for process kube-apiserver to stop

Yeah, I'll try to figure out what's going on here...

File: .github/workflows/presubmit.yaml (hunk 1 above)
Comment (diviner524) [reply .github/workflows/presubmit.yaml:31]: Gotcha. I use VSCode, but I'm cool with whatever works!

**Comment by justinsb** [comment]
Flakes look to be fixed by bumping the shutdown timeout to 1 minute, done in #1030

**Review by jingyih — COMMENTED** [review]
/lgtm

**Pull request merged**